packages/react-native/Libraries/Components/StatusBar/StatusBar.js

```diff
@@ -417,6 +418,14 @@ class StatusBar extends React.Component<Props> {
    * Updates the native status bar with the props from the stack.
    */
   static _updatePropsStack = () => {
+    if (Platform.isVisionOS) {
+      warnOnce(
+        'StatusBar-unavailable',
+        'StatusBar is not available on visionOS platform.',
+      );
+      return;
+    }
```

Comment:
Let's move this to `componentDidMount()`

Reply:
Done!